### PR TITLE
refactor: update logic for `isAbsoluteURL` method

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

### Motivation / Use-Case

Fix https://github.com/webpack/webpack-dev-server/issues/4037
Fix https://github.com/webpack/webpack-dev-server/pull/4038

### Breaking Changes

None

### Additional Info
No

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -50,15 +50,18 @@ class Server {
...
+      const isWindowsPath = /^[a-zA-Z]:\\/.test(uri);
+
+      return !isWindowsPath;
+    } catch (error) {
       return false;
     }
-
-    // Scheme: https://tools.ietf.org/html/rfc3986#section-3.1
-    // Absolute URL: https://tools.ietf.org/html/rfc3986#section-4.3
-    return /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(URL);
```
Comment (alexander-akait) [inline lib/Server.js:61]: To be honestly I don't think it is faster :+1: Before we have one simple regexp, now we have `new URL(...)` and regexp...

Comment (snitin315) [reply lib/Server.js:61]: Hmm, So, how should we proceed here? Do you have any alternative in mind? Or should we close the related issue?

Comment (alexander-akait) [reply lib/Server.js:61]: I think we can keep code as is and do not modify it, it is not bottleneck for us

Comment (snitin315) [reply lib/Server.js:61]: Let's close it then 👍

Comment (alexander-akait) [reply lib/Server.js:61]: Yep, maybe we revisit it in future, but I don't see real problems with our check right now, we don't need to parse whole URL, we just need to check string is absolute URL

**Pull request closed without merging**